tests/Integration/Schema/Directives/CanDirectiveDBTest.php

```diff
@@ -287,4 +288,239 @@ public function testWorksWithSoftDeletes(): void
+                name
+            }
+        }
+        ')->assertJson([
+            'data' => [
+                'user' => null,
+            ],
+            'errors' => [
+                [
+                    'message' => 'No query results for model [Tests\Utils\Models\User].',
+                ],
+            ],
+        ]);
+    }
+
+    public function testHandleMultipleModelsWithFindByArgs(): void
+    {
+        $user = new User();
+        $user->name = UserPolicy::ADMIN;
+        $this->be($user);
+
+        $postA = factory(Post::class)->create([
+            'user_id' => $user->getKey(),
+            'title' => 'Harry Potter and the Half-Blood Prince',
+        ]);
+        $postB = factory(Post::class)->create([
+            'user_id' => $user->getKey(),
+            'title' => 'Harry Potter and the Chamber of Secrets',
+        ]);
+
+        $this->schema = /** @lang GraphQL */ '
+        type Query {
+            deletePosts(ids: [ID!]!): [Post!]!
+                @can(ability: "delete", findByArgs: true)
+                @delete
+        }
+
+        type Post {
+            id: ID!
+            title: String!
+        }
+        ';
+
+        $this->graphQL(/** @lang GraphQL */ "
+        {
+            deletePosts(ids: [{$postA->getKey()}, {$postB->getKey()}]) {
+                title
+            }
+        }
+        ")->assertJson([
+            'data' => [
+                'deletePosts' => [
+                    [
+                        'title' => 'Harry Potter and the Half-Blood Prince',
+                    ],
+                    [
+                        'title' => 'Harry Potter and the Chamber of Secrets',
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    public function testWorksWithSoftDeletesWithFindByArgs(): void
+    {
+        $user = new User();
+        $user->name = UserPolicy::ADMIN;
+        $this->be($user);
+
+        $task = factory(Task::class)->create();
+        $task->delete();
+
+        $this->schema = /** @lang GraphQL */ '
+        type Query {
+            task(id: ID @eq): Task
+                @can(ability: "adminOnly", findByArgs: true)
+                @softDeletes
+                @find
+        }
+
+        type Task {
+            name: String!
+        }
+        ';
+
+        $this->graphQL(/** @lang GraphQL */ "
+        {
+            task(id: {$task->getKey()}, trashed: WITH) {
+                name
+            }
+        }
+        ")->assertJson([
+            'data' => [
+                'task' => [
+                    'name' => $task->name,
+                ],
+            ],
+        ]);
+    }
+
+    public function testFindAndFindByArgsValidation(): void
+    {
+        $user = new User();
+        $user->name = UserPolicy::ADMIN;
+        $this->be($user);
+
+        $user = factory(User::class)->create([
+            'name' => 'foo',
+        ]);
+
+        $this->schema = /** @lang GraphQL */ '
```

Comment:
```suggestion
        $this->buildSchema(/** @lang GraphQL */ '
```

Validated at build time, no need to fire a query.